zkvm/src/vm.rs

```diff
@@ -36,6 +39,32 @@ pub struct TxHeader {
     pub maxtime: u64,
 }
 
+impl TxHeader {
+    fn serialized_size(&self) -> usize {
+        8 * 3
+    }
+
+    fn encode(&self, buf: &mut Vec<u8>) {
+        encoding::write_u64(self.version, buf);
+        encoding::write_u64(self.mintime, buf);
+        encoding::write_u64(self.maxtime, buf);
+    }
+
+    fn decode<'a>(reader: &mut SliceReader<'a>) -> Result<Self, VMError> {
+        let (header, _) = reader.slice(|r| {
```

Comment:
`.slice` method is only needed when you need the surrounding slice, e.g. to hash the serialized object w/o re-serializing it. Here we do not need that slice (hence use of `_` in this line), so the whole `.slice` call is not needed. We can directly do `reader.read_u64()` etc.

Reply:
```rust
fn decode(reader: ...) {
    Ok(TxHeader {
        version: r.read_u64()?,
        mintime: r.read_u64()?,
        maxtime: r.read_u64()?,
    })
}
```

Reply:
Got it, this was super helpful, thank you!